src/main/java/com/group2/catanbackend/gamelogic/GameLogicController.java

```diff
@@ -163,6 +174,99 @@ private boolean stealResource(Player playerToStealFrom, Player playerToGiveTo) {
         }
         return false;
     }
+    //TODO: error messages @daniel
+    private void makeAcceptMove(AcceptMoveDto acceptMove, Player player){
+        if(isSetupPhase)
+            throw new NotActivePlayerException(ErrorCode.ERROR_NOT_ACTIVE_PLAYER.formatted(activePlayer.getDisplayName()));
+        if(currentTrade==null)//trade is gone
+            throw new InvalidGameMoveException(ErrorCode.ERROR_TRADE_NOT_AVAILABLE);
+        if(activePlayer.getInGameID()!=acceptMove.getTradeOfferDto().getFromPlayer().getInGameID())
+            throw new InvalidGameMoveException(ErrorCode.ERROR_NOT_ENOUGH_RESOURCES.formatted("active player"));
+        if(!currentTrade.equals(acceptMove.getTradeOfferDto()))
+            throw new InvalidGameMoveException(ErrorCode.ERROR_NOT_ENOUGH_RESOURCES.formatted("current trade not accepting trade"));//TradeOfferDtos are not the same
+        if (!player.resourcesSufficient(acceptMove.getTradeOfferDto().getGiveResources()))
+            throw new InvalidGameMoveException(ErrorCode.ERROR_NOT_ENOUGH_RESOURCES.formatted("the REAL resources sufficient"));
+        if(!activePlayer.resourcesSufficient(negateAllValues(acceptMove.getTradeOfferDto().getGetResources())))
+            throw new InvalidGameMoveException(ErrorCode.ERROR_NOT_ENOUGH_RESOURCES.formatted("receiving player resources"));//trader does not have enough resources
+        computeAcceptMove(acceptMove.getTradeOfferDto(), player);
+    }
+    private void computeAcceptMove(TradeOfferDto tradeOffer, Player player){//more
+        activePlayer.adjustResources(negateAllValues(tradeOffer.getGetResources()));
+        activePlayer.adjustResources(negateAllValues(tradeOffer.getGiveResources()));
+        player.adjustResources(tradeOffer.getGiveResources());
+        player.adjustResources(tradeOffer.getGetResources());
+        this.currentTrade=null;
+        sendCurrentGameStateToPlayers();
+        //send new GameProgressDto? but with what content?
+    }
+
+    //TODO: rethink checks. I did not until now
+    private void makeTradeMove(TradeMoveDto tradeMove, Player player){
+        //if (isSetupPhase) throw new InvalidGameMoveException(ErrorCode.ERROR_CANT_ROLL_IN_SETUP);
+        if (activePlayer != player)
+            throw new NotActivePlayerException(ErrorCode.ERROR_NOT_ACTIVE_PLAYER.formatted(activePlayer.getDisplayName()));
+        if(tradeMove.getToPlayers().size() != players.size()-1)
```

Comment:
doesnt this always fail if: 
- you dont want to send the trade to every player
- you want to trade with the bank
as then the getToPlayers list is not the same size as the players list -1?